tests/phpunit/tests/l10n/wpLocaleSwitcher.php

```diff
@@ -27,7 +27,7 @@ public function set_up() {
 		/** @var WP_Textdomain_Registry $wp_textdomain_registry */
 		global $wp_textdomain_registry;
 
-		$wp_textdomain_registry->reset();
+		$wp_textdomain_registry = new WP_Textdomain_Registry();
```

Comment:
I love to see a test that proves that `cached_mo_files` value is populated and on the second run, `is_readable` is not called. A test when the list exist and when the file does not exist would be great.

Reply:
The `test_get_locale_is_called_only_once_per_textdomain_with_custom_lang_dir` test is _exactly_ the one covering this from a to z.